src/Orleans.Runtime/Core/Dispatcher.cs

```diff
@@ -652,21 +294,9 @@ void OnAddressingFailure(Message m, IGrainContext activation, Exception ex)
             }
         }
 
-        // this is a compatibility method for portions of the code base that don't use
-        // async/await yet, which is almost everything. there's no liability to discarding the
-        // Task returned by AsyncSendMessage()
-        internal void SendMessage(Message message, IGrainContext sendingActivation = null)
-        {
-            AsyncSendMessage(message, sendingActivation).Ignore();
-        }
-
         /// <summary>
         /// Resolve target address for a message
-        /// - use transaction info
-        /// - check ordering info in message and sending activation
-        /// - use sender's placement strategy
         /// </summary>
-        /// <param name="message"></param>
         /// <returns>Resolve when message is addressed (modifies message fields)</returns>
         private Task AddressMessage(Message message)
```

Comment:
Shoould we change this method to return a `ValueTask` instead?

Reply:
I think we can leave it for now because it's an untyped task (no return value) so the synchronous path can return `Task.CompletedTask`

Reply:
We *should* convert `IActivationSelector`. I can do that in this PR if you're ok with it

Reply:
Added, it's not too much change